File: algorithm/Dijkstra/dijkstra.c
// add graph undirect (undirect is line haven't pointer x ->)
#include<stdio.h>
#define MAX 9
#define INF 9999
int graph[MAX][MAX] = {{0, 4, 0, 0, 0, 0, 0, 8, 0},
                      {4, 0, 8, 0, 0, 0, 0, 11, 0},
                      {0, 8, 0, 7, 0, 4, 0, 0, 2},
                      {0, 0, 7, 0, 9, 14, 0, 0, 0},
                      {0, 0, 0, 9, 0, 10, 0, 0, 0},
                      {0, 0, 4, 14, 10, 0, 2, 0, 0},
                      {0, 4, 0, 0, 0, 2, 0, 1, 6},
                      {8, 11, 0, 0, 0, 0, 1, 0, 7},
                      {0, 0, 2, 0, 0, 0, 6, 7, 0}};

int main() {
  int i,j,vertex,source[MAX],select[MAX],min,index_min;
  printf("input start vertex : ");
  scanf("%d",&vertex);
  for(i = 0;i<MAX;i++) {
    if(graph[vertex][i] != 0 || vertex == i) {
      source[i] = graph[vertex][i];
    } else {
      source[i] = INF;
    }
    if(i == vertex) {
      select[i] = 1;
    } else {
      select[i] = 0;
    }
  }
  for(i = 0;i<MAX-1;i++) {
    min = INF;
    for(j = 0;j<MAX;j++) {
      if(source[j] < min && select[j] == 0) {
        min = source[j];
        index_min = j;
      }
    }
    select[index_min] = 1;
    for(j = 0;j<MAX;j++) {
      if(graph[index_min][j] + min < source[j] && graph[index_min][j] != 0) {
        source[j] = graph[index_min][j] + min;
      }
    }
  }
  for(i = 0;i<MAX;i++) {
    printf("source : %d ---> des %d : value : %d\n",vertex,i,source[i]);
  }
}

File: algorithm/Dijkstra/dijkstra.js
const graph = [
  [0, 8, 2, 5, 0, 0, 0, 0],
  [8, 0, 0, 2, 0, 13, 0, 0],
  [2, 0, 0, 2, 5, 0, 0, 0],
  [5, 2, 2, 0, 1, 6, 3, 0],
  [0, 0, 5, 1, 0, 0, 1, 0],
  [0, 13, 0, 6, 0, 0, 2, 3],
  [0, 0, 0, 3, 1, 2, 0, 6],
  [0, 0, 0, 0, 0, 6, 3, 0],
];

const show = new Array();

let select = [],
  answer = [],
  min = Infinity,
  choose = 5,
  u = [],
  v = [],
  V = graph.length,
  show_index = 0;

for (let i = 0; i < V; i++) {
  select[i] = 0;
  show[i] = choose;
  for (let j = 0; j < V; j++) {
    if (i !== j && graph[i][j] === 0) {
      graph[i][j] = Infinity;
    }
  }
}

select[choose] = 1;
u[choose] = 0;

for (let i = 0; i < V; i++) {
  v[i] = graph[choose][i];
}

for (let i = 0; i < V; i++) {
  select[choose] = 1;
  min = Infinity;
  for (let j = 0; j < V; j++) {
    if (u[choose] + graph[choose][j] < v[j] && choose != j) {
      v[j] = u[choose] + graph[choose][j];
      show[j] = choose;
    }
  }

  for (let j = 0; j < V; j++) {
    if (v[j] < min && choose !== j && select[j] === 0) {
      min = v[j];
      choose = j;
      u[j] = min;
    }
  }
  console.log(v);
}
console.log("------------------");
console.log(show);